torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,293 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""L7 Irish dataset."""
+
+import glob
+import os
+import re
+from typing import Any, Callable, Dict, List, Optional, Sequence, cast
+
+import matplotlib.pyplot as plt
+import torch
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class L7Irish(RasterDataset):
+    """L7 Irish dataset.
+
+    The `L7 Irish <https://www.sciencebase.gov/catalog/item/573ccf18e4b0dae0d5e4b109>`__ dataset
```

Comment:
```suggestion
    The `L7 Irish <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset
```
Let's link to this site instead. It has a more accurate description of the mask classes and loads faster.